### ISSUE TITLE
Bokeh and matplotlib bar plots not consistent

### ISSUE THREAD

**Issue by redst4r** [ioam/holoviews#862]
Just some strange behavior I found while working through this [tutorial](http://holoviews.org/Tutorials/Columnar_Data.html), specifically the aggregation/reduce part.
Working with the matplotlib backend, plots turn out fine

``` python
import holoviews as hv
import numpy as np
hv.notebook_extension()

n = np.arange(1000)
xs = np.repeat(range(2), 500)
ys = n%4
zs = np.random.randn(1000)
table = hv.Table((xs, ys, zs), kdims=['x', 'y'], vdims=['z'])

hv.Bars(table).aggregate(function=np.mean) + hv.Bars(table).reduce(x=np.mean)
```

That gives this plot ![plot](https://imgupload.in/images/2016/09/15/normal.png)

Runnning the same with bokeh backend

``` python
hv.notebook_extension('bokeh')
```

gives this strange result ![result](https://imgupload.in/images/2016/09/15/Screenshotfrom2016-09-1510-41-08.png)

No idea whats going on.
- Holoviews version 1.6.2
- bokeh version 0.12.0
- python 3.5.2

**Comment by jbednar** [comment]
The bokeh team is strongly considering splitting out the Charts interface as a separate package before the 1.0 release, to help focus the core into a manageable library.  It would then make sense for HoloViews to rely only on the main Bokeh library, because in some sense HoloViews is an alternative API at the same level as the Charts API, building on top of the Bokeh core to provide a nicer high-level set of capabilities.  But the HoloViews API is a lot more powerful than Charts in its current form.

**Comment by philippjfr** [comment]
At SciPy we sat in some discussion surrounding the bokeh charts API and altair, during which it was considered to provide a path to go straight from altair to a bokeh plot, which would supplant the charts API entirely. Either way building all our Elements on the lower level plotting interface seems like a good idea.

**Comment by philippjfr** [comment]
Fixed in #1407, closing.
